lib/elixir/lib/module.ex

```diff
@@ -1221,12 +1221,75 @@ defmodule Module do
     behaviours = :ets.lookup_element(table, :behaviour, 2)
     impls = :ets.lookup_element(table, {:elixir, :impls}, 2)
 
+    check_behaviours(env, behaviours, all_definitions)
+
     if impls != [] do
       non_implemented_callbacks = check_impls(behaviours, impls)
       warn_missing_impls(env, non_implemented_callbacks, all_definitions, overridable_pairs)
     end
   end
 
+  defp check_behaviours(env, behaviours, all_definitions) do
+    Enum.each(behaviours, fn behaviour ->
+      cond do
+        is_standard_behaviour(behaviour) ->
+          nil
+        not is_atom(behaviour) ->
+          :elixir_errors.warn(env.line, env.file, "@behaviour #{inspect behaviour} must be an atom (for module #{inspect env.module})")
```

Comment:
Maybe "(in module #{inspect ...})" here and below would read better?

Reply:
It would.  I am making the change.